### PR TITLE
[MRG] API for Static Topic for word in Vocabulary.

### PR DESCRIPTION
**Pull request by bhargavvader**
@tmylk to complement the dynamic word document topic, this is a PR for static word topics.

### CODE REVIEW

**Comment by tmylk** [comment]
Thanks. Can the output be in same format as [`get_document_topics`](
https://github.com/piskvorky/gensim/blob/48105777b4c162c7e9d3f4051e6232c384177cd0/gensim/models/ldamodel.py#L886)?

**Comment by bhargavvader** [comment]
Would that mean a tuple with that word and corresponding topic?

**Comment by bhargavvader** [comment]
@tmylk made it the same format as `get_document_topic`. Again, `glove2word2vec` fails. How do I reset it so that it passes? The static word tests work fine.

File: gensim/models/ldamodel.py [hunk 1]
```diff
@@ -905,6 +905,28 @@ def get_document_topics(self, bow, minimum_probability=None):
         return [(topicid, topicvalue) for topicid, topicvalue in enumerate(topic_dist)
                 if topicvalue >= minimum_probability]
 
+    def get_static_topic(self, word_id, minimum_probability=None):
```
Comment (tmylk) [inline gensim/models/ldamodel.py]: Default minimum_probability=0 is most intuitive

File: gensim/test/test_ldamodel.py [hunk 2]
```diff
@@ -265,6 +265,39 @@ def testGetDocumentTopics(self):
...
+            # check with id 
+            result = model.get_static_topic(2)
+            expected = [(1, 0.10658834157061754)]
+            self.assertEqual(result[0][0], expected[0][0])
+            self.assertAlmostEqual(result[0][1], expected[0][1])
+
+
+            # if user has entered word instead, check with word
+            result = model.get_static_topic(str(model.id2word[2]))
+            expected = [(1, 0.10658834157061754)]
```
Comment (tmylk) [inline gensim/test/test_ldamodel.py]: The precision of `0.10` is ok. That way multicore and singlecore models should have the same test.

File: gensim/models/ldamodel.py (hunk 1 above)
Comment (piskvorky) [inline gensim/models/ldamodel.py]: I disagree. `0.0` is a specific value to use, whereas `None` stands for "use default". Different semantics.

File: gensim/test/test_ldamodel.py (hunk 2 above)
Comment (bhargavvader) [inline gensim/test/test_ldamodel.py]: Cool. Changed it.

**Comment by bhargavvader** [comment]
@tmylk , @piskvorky , could you please review? I've finished and wrote tests.

File: gensim/models/ldamodel.py [hunk 3]
```diff
@@ -905,6 +905,28 @@ def get_document_topics(self, bow, minimum_probability=None):
...
+        """
+        Returns static topics for word in vocabulary.
```
Comment (piskvorky) [inline gensim/models/ldamodel.py]: Needs better docs -- what is a "static topic"?

File: gensim/test/test_ldamodel.py [hunk 4]
```diff
@@ -265,6 +265,24 @@ def testGetDocumentTopics(self):
...
+
+        numpy.random.seed(0)
+        model = self.class_(self.corpus, id2word=dictionary, num_topics=2, passes= 100)
+
+        # check with id 
+        result = model.get_static_topic(2)
+        expected = [(1, 0.1066)]
+        self.assertEqual(result[0][0], expected[0][0])
+        self.assertAlmostEqual(result[0][1], expected[0][1], places=2)
+
```
Comment (piskvorky) [inline gensim/test/test_ldamodel.py]: PEP8 (too many blank lines).

File: gensim/test/test_ldamodel.py [hunk 5]
```diff
@@ -265,6 +265,24 @@ def testGetDocumentTopics(self):
             self.assertTrue(isinstance(k, int))
             self.assertTrue(isinstance(v, float))
 
+    def testStaticTopics(self):
+
+        numpy.random.seed(0)
+        model = self.class_(self.corpus, id2word=dictionary, num_topics=2, passes= 100)
```
Comment (piskvorky) [inline gensim/test/test_ldamodel.py]: PEP8: no spaces around argument params.

**Comment by piskvorky** [comment]
The method name doesn't seem right; @cscorley any better ideas?

Maybe `get_word_topics()`?

**Comment by bhargavvader** [comment]
@piskvorky , yeah `get_term_topics()` does sound better.

**Comment by piskvorky** [comment]
Looks good, thanks @bhargavvader .

I leave the final sanity checks & merge touches to @tmylk

**Comment by tmylk** [comment]
Just a note in the changelog and then ready to merge.

**Comment by bhargavvader** [comment]
@tmylk done!

**Comment by cscorley** [comment]
Yep, `get_term_topics` sounds great for this.

**Comment by piskvorky** [comment]
@tmylk when are planning to release? This version is getting scarily bulky again.

Does it make sense to release before the PyCon coding sprint, so people can work off a fresh version?

**Pull request merged**